addons/website_sale/models/product.py

```diff
@@ -207,6 +208,43 @@ class ProductTemplate(models.Model):
 
     product_template_image_ids = fields.One2many('product.image', 'product_tmpl_id', string="Extra Product Media", copy=True)
 
+    uom_count = fields.Float('Base Unit Quantity', required=True, default=1)
+    uom_char = fields.Char('Sales Unit to Display')
+    price_per_unit_display_string = fields.Char("PPU to Display", compute='_compute_price_per_unit_string')
+
+    def _compute_price_per_uom(self):
+        for template in self:
+            if template.price:
+                return template.price / template.uom_count if template.uom_count else 0
+            else:
+                return template.lst_price / template.uom_count if template.uom_count else 0
+
+    @api.depends('uom_id', 'uom_char', 'uom_count', 'price', 'lst_price')
+    def _compute_price_per_unit_string(self):
+        for template in self:
+            ppu = template._compute_price_per_uom()
+            if ppu:
+                if template.uom_char:
+                    template.price_per_unit_display_string = f"({template.currency_id.symbol} " \
+                                                             f"{float_repr(ppu, 2)} / " \
+                                                             f"{template.uom_char})"
+                else:
+                    template.price_per_unit_display_string = f"({template.currency_id.symbol} " \
+                                                             f"{float_repr(ppu, 2)} / " \
+                                                             f"{template.uom_id.name})"
```

Comment:
```suggestion
                if template.uom_char:
                template.price_per_unit_display_string = f"({template.currency_id.symbol} " \
                                                             f"{float_repr(ppu, 2)} / " \
                                                             f"{template.uom_char or template.uom_id.name})"
```

2? use the precision of the db. symbol order ? why not just use a monetary which will use correct precision, symbol position, ...  ?